src/mon/FSCommands.cc

```diff
@@ -385,6 +384,16 @@ class SetHandler : public FileSystemCommandHandler
       return -EINVAL;
     }
 
+  bool confirm = false;
+  cmd_getval(cmdmap, "yes_i_really_mean_it", confirm);
+  if (var == "max_mds" && !confirm && mon->mdsmon()->has_health_warnings()) {
+    ss << "FS has health warnings and modifying value of FS setting variable "
+       << "'max_mds' will not help troubleshoot or recover from one or more "
+       << "health warnings. If you wish to proceed, run again with "
+       << "--yes-i-really-mean-it";
+    return -EPERM;
+  }
+
```

Comment:
Needs doc update for `--yes-i-really-mean-it`

Reply:
Done.